### PR TITLE
Add fsp module

### PR DESCRIPTION
**Pull request by lispandfound**
This pull request adds an FSP module to read the FSP file format that is commonly used in SRCMOD to represent rupture inversion model outputs. See the SRCMOD website for a [description of the FSP format](http://equake-rc.info/SRCMOD/fileformats/fsp/).

### CODE REVIEW

File: source_modelling/fsp.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                # Strip the leading "% ", and deduplicate the spaces in the line.
+                # This is required to normalise the string so that the parse
+                # module can handle the rest of the parsing
+                header += " ".join(line.lstrip("% ").split()) + "\n"
+
+            # Now we can parse the header according to the header pattern.
+            parse_result = parse.parse(
+                HEADER_PATTERN,
+                header,
+            )
```
Comment (joelridden) [inline source_modelling/fsp.py:208]: I like this

**Review by joelridden — APPROVED** [review]

File: source_modelling/fsp.py (hunk 1 above)
Comment (lispandfound) [reply source_modelling/fsp.py:208]: Yeah the parse package is really nice, good alternative to regex for simple use cases.

**Review by AndrewRidden-Harper — APPROVED** [review]
nice

**Review by claudio525 — APPROVED** [review]

File: source_modelling/fsp.py (hunk 1 above)
Comment (sungeunbae) [reply source_modelling/fsp.py:208]: Finally we have a `scanf ` alternative.

**Review by sungeunbae — APPROVED** [review]

**Pull request merged**